daemon/k8s_watcher.go

```diff
@@ -1486,6 +1555,12 @@ func (d *Daemon) deleteCiliumNetworkPolicyV1(cnp *cilium_v1.CiliumNetworkPolicy)
 
 	scopedLog.Debug("Deleting CiliumNetworkPolicy")
 
+	ctrlName := cnp.GetControllerName()
+	err := cnpCM.RemoveController(ctrlName)
```

Comment:
Can this still race with an UpdateController call? I'm guessing no since the Informer gives us events in order (and we do delete/create in the update case). If it does race we would still need something like https://github.com/cilium/cilium/pull/2895 but then you would have to store the controller reference more globally

Reply:
If it does we have a bigger problem by deleting the CNP